// @react-spectrum/utils/dist/styleProps.main.js
var $893a66ba513f4a66$exports = require("./BreakpointProvider.main.js");
var $5XTxG$reactariai18n = require("@react-aria/i18n");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "baseStyleProps", () => $d3b73be57066120b$export$fe9c6e915565b4e8);
$parcel$export(module.exports, "dimensionValue", () => $d3b73be57066120b$export$abc24f5b99744ea6);
$parcel$export(module.exports, "passthroughStyle", () => $d3b73be57066120b$export$46b6c81d11d2c30a);
$parcel$export(module.exports, "viewStyleProps", () => $d3b73be57066120b$export$e0705d1a55f297c);
$parcel$export(module.exports, "responsiveDimensionValue", () => $d3b73be57066120b$export$f348bec194f2e6b5);
$parcel$export(module.exports, "getResponsiveProp", () => $d3b73be57066120b$export$52dbfdbe1b2c3541);
$parcel$export(module.exports, "convertStyleProps", () => $d3b73be57066120b$export$f3c39bb9534218d0);
$parcel$export(module.exports, "useStyleProps", () => $d3b73be57066120b$export$b8e6fb9d2dff3f41);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

const $d3b73be57066120b$export$fe9c6e915565b4e8 = {
    margin: [
        'margin',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    marginStart: [
        $d3b73be57066120b$var$rtl('marginLeft', 'marginRight'),
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    marginEnd: [
        $d3b73be57066120b$var$rtl('marginRight', 'marginLeft'),
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    // marginLeft: ['marginLeft', dimensionValue],
    // marginRight: ['marginRight', dimensionValue],
    marginTop: [
        'marginTop',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    marginBottom: [
        'marginBottom',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    marginX: [
        [
            'marginLeft',
            'marginRight'
        ],
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    marginY: [
        [
            'marginTop',
            'marginBottom'
        ],
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    width: [
        'width',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    height: [
        'height',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    minWidth: [
        'minWidth',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    minHeight: [
        'minHeight',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    maxWidth: [
        'maxWidth',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    maxHeight: [
        'maxHeight',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    isHidden: [
        'display',
        $d3b73be57066120b$var$hiddenValue
    ],
    alignSelf: [
        'alignSelf',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    justifySelf: [
        'justifySelf',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    position: [
        'position',
        $d3b73be57066120b$var$anyValue
    ],
    zIndex: [
        'zIndex',
        $d3b73be57066120b$var$anyValue
    ],
    top: [
        'top',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    bottom: [
        'bottom',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    start: [
        $d3b73be57066120b$var$rtl('left', 'right'),
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    end: [
        $d3b73be57066120b$var$rtl('right', 'left'),
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    left: [
        'left',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    right: [
        'right',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    order: [
        'order',
        $d3b73be57066120b$var$anyValue
    ],
    flex: [
        'flex',
        $d3b73be57066120b$var$flexValue
    ],
    flexGrow: [
        'flexGrow',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    flexShrink: [
        'flexShrink',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    flexBasis: [
        'flexBasis',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    gridArea: [
        'gridArea',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    gridColumn: [
        'gridColumn',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    gridColumnEnd: [
        'gridColumnEnd',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    gridColumnStart: [
        'gridColumnStart',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    gridRow: [
        'gridRow',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    gridRowEnd: [
        'gridRowEnd',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ],
    gridRowStart: [
        'gridRowStart',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ]
};
const $d3b73be57066120b$export$e0705d1a55f297c = {
    ...$d3b73be57066120b$export$fe9c6e915565b4e8,
    backgroundColor: [
        'backgroundColor',
        $d3b73be57066120b$var$backgroundColorValue
    ],
    borderWidth: [
        'borderWidth',
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderStartWidth: [
        $d3b73be57066120b$var$rtl('borderLeftWidth', 'borderRightWidth'),
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderEndWidth: [
        $d3b73be57066120b$var$rtl('borderRightWidth', 'borderLeftWidth'),
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderLeftWidth: [
        'borderLeftWidth',
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderRightWidth: [
        'borderRightWidth',
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderTopWidth: [
        'borderTopWidth',
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderBottomWidth: [
        'borderBottomWidth',
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderXWidth: [
        [
            'borderLeftWidth',
            'borderRightWidth'
        ],
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderYWidth: [
        [
            'borderTopWidth',
            'borderBottomWidth'
        ],
        $d3b73be57066120b$var$borderSizeValue
    ],
    borderColor: [
        'borderColor',
        $d3b73be57066120b$var$borderColorValue
    ],
    borderStartColor: [
        $d3b73be57066120b$var$rtl('borderLeftColor', 'borderRightColor'),
        $d3b73be57066120b$var$borderColorValue
    ],
    borderEndColor: [
        $d3b73be57066120b$var$rtl('borderRightColor', 'borderLeftColor'),
        $d3b73be57066120b$var$borderColorValue
    ],
    borderLeftColor: [
        'borderLeftColor',
        $d3b73be57066120b$var$borderColorValue
    ],
    borderRightColor: [
        'borderRightColor',
        $d3b73be57066120b$var$borderColorValue
    ],
    borderTopColor: [
        'borderTopColor',
        $d3b73be57066120b$var$borderColorValue
    ],
    borderBottomColor: [
        'borderBottomColor',
        $d3b73be57066120b$var$borderColorValue
    ],
    borderXColor: [
        [
            'borderLeftColor',
            'borderRightColor'
        ],
        $d3b73be57066120b$var$borderColorValue
    ],
    borderYColor: [
        [
            'borderTopColor',
            'borderBottomColor'
        ],
        $d3b73be57066120b$var$borderColorValue
    ],
    borderRadius: [
        'borderRadius',
        $d3b73be57066120b$var$borderRadiusValue
    ],
    borderTopStartRadius: [
        $d3b73be57066120b$var$rtl('borderTopLeftRadius', 'borderTopRightRadius'),
        $d3b73be57066120b$var$borderRadiusValue
    ],
    borderTopEndRadius: [
        $d3b73be57066120b$var$rtl('borderTopRightRadius', 'borderTopLeftRadius'),
        $d3b73be57066120b$var$borderRadiusValue
    ],
    borderBottomStartRadius: [
        $d3b73be57066120b$var$rtl('borderBottomLeftRadius', 'borderBottomRightRadius'),
        $d3b73be57066120b$var$borderRadiusValue
    ],
    borderBottomEndRadius: [
        $d3b73be57066120b$var$rtl('borderBottomRightRadius', 'borderBottomLeftRadius'),
        $d3b73be57066120b$var$borderRadiusValue
    ],
    borderTopLeftRadius: [
        'borderTopLeftRadius',
        $d3b73be57066120b$var$borderRadiusValue
    ],
    borderTopRightRadius: [
        'borderTopRightRadius',
        $d3b73be57066120b$var$borderRadiusValue
    ],
    borderBottomLeftRadius: [
        'borderBottomLeftRadius',
        $d3b73be57066120b$var$borderRadiusValue
    ],
    borderBottomRightRadius: [
        'borderBottomRightRadius',
        $d3b73be57066120b$var$borderRadiusValue
    ],
    padding: [
        'padding',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    paddingStart: [
        $d3b73be57066120b$var$rtl('paddingLeft', 'paddingRight'),
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    paddingEnd: [
        $d3b73be57066120b$var$rtl('paddingRight', 'paddingLeft'),
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    paddingLeft: [
        'paddingLeft',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    paddingRight: [
        'paddingRight',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    paddingTop: [
        'paddingTop',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    paddingBottom: [
        'paddingBottom',
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    paddingX: [
        [
            'paddingLeft',
            'paddingRight'
        ],
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    paddingY: [
        [
            'paddingTop',
            'paddingBottom'
        ],
        $d3b73be57066120b$export$abc24f5b99744ea6
    ],
    overflow: [
        'overflow',
        $d3b73be57066120b$export$46b6c81d11d2c30a
    ]
};
const $d3b73be57066120b$var$borderStyleProps = {
    borderWidth: 'borderStyle',
    borderLeftWidth: 'borderLeftStyle',
    borderRightWidth: 'borderRightStyle',
    borderTopWidth: 'borderTopStyle',
    borderBottomWidth: 'borderBottomStyle'
};
function $d3b73be57066120b$var$rtl(ltr, rtl) {
    return (direction)=>direction === 'rtl' ? rtl : ltr;
}
const $d3b73be57066120b$var$UNIT_RE = /(%|px|em|rem|vw|vh|auto|cm|mm|in|pt|pc|ex|ch|rem|vmin|vmax|fr)$/;
const $d3b73be57066120b$var$FUNC_RE = /^\s*\w+\(/;
const $d3b73be57066120b$var$SPECTRUM_VARIABLE_RE = /(static-)?size-\d+|single-line-(height|width)/g;
function $d3b73be57066120b$export$abc24f5b99744ea6(value) {
    if (typeof value === 'number') return value + 'px';
    if (!value) return undefined;
    if ($d3b73be57066120b$var$UNIT_RE.test(value)) return value;
    if ($d3b73be57066120b$var$FUNC_RE.test(value)) return value.replace($d3b73be57066120b$var$SPECTRUM_VARIABLE_RE, 'var(--spectrum-global-dimension-$&, var(--spectrum-alias-$&))');
    return `var(--spectrum-global-dimension-${value}, var(--spectrum-alias-${value}))`;
}
function $d3b73be57066120b$export$f348bec194f2e6b5(value, matchedBreakpoints) {
    value = $d3b73be57066120b$export$52dbfdbe1b2c3541(value, matchedBreakpoints);
    return $d3b73be57066120b$export$abc24f5b99744ea6(value);
}
function $d3b73be57066120b$var$colorValue(value, type = 'default', version = 5) {
    if (version > 5) return `var(--spectrum-${value}, var(--spectrum-semantic-${value}-color-${type}))`;
    return `var(--spectrum-legacy-color-${value}, var(--spectrum-global-color-${value}, var(--spectrum-semantic-${value}-color-${type})))`;
}
function $d3b73be57066120b$var$backgroundColorValue(value, version = 5) {
    if (!value) return undefined;
    return `var(--spectrum-alias-background-color-${value}, ${$d3b73be57066120b$var$colorValue(value, 'background', version)})`;
}
function $d3b73be57066120b$var$borderColorValue(value, version = 5) {
    if (!value) return undefined;
    if (value === 'default') return 'var(--spectrum-alias-border-color)';
    return `var(--spectrum-alias-border-color-${value}, ${$d3b73be57066120b$var$colorValue(value, 'border', version)})`;
}
function $d3b73be57066120b$var$borderSizeValue(value) {
    return value && value !== 'none' ? `var(--spectrum-alias-border-size-${value})` : '0';
}
function $d3b73be57066120b$var$borderRadiusValue(value) {
    if (!value) return undefined;
    return `var(--spectrum-alias-border-radius-${value})`;
}
function $d3b73be57066120b$var$hiddenValue(value) {
    return value ? 'none' : undefined;
}
function $d3b73be57066120b$var$anyValue(value) {
    return value;
}
function $d3b73be57066120b$var$flexValue(value) {
    if (typeof value === 'boolean') return value ? '1' : undefined;
    return '' + value;
}
function $d3b73be57066120b$export$f3c39bb9534218d0(props, handlers, direction, matchedBreakpoints) {
    let style = {};
    for(let key in props){
        let styleProp = handlers[key];
        if (!styleProp || props[key] == null) continue;
        let [name, convert] = styleProp;
        if (typeof name === 'function') name = name(direction);
        let prop = $d3b73be57066120b$export$52dbfdbe1b2c3541(props[key], matchedBreakpoints);
        let value = convert(prop, props.colorVersion);
        if (Array.isArray(name)) for (let k of name)style[k] = value;
        else style[name] = value;
    }
    for(let prop in $d3b73be57066120b$var$borderStyleProps)if (style[prop]) {
        style[$d3b73be57066120b$var$borderStyleProps[prop]] = 'solid';
        style.boxSizing = 'border-box';
    }
    return style;
}
function $d3b73be57066120b$export$b8e6fb9d2dff3f41(props, handlers = $d3b73be57066120b$export$fe9c6e915565b4e8, options = {}) {
    let { UNSAFE_className: UNSAFE_className, UNSAFE_style: UNSAFE_style, ...otherProps } = props;
    let breakpointProvider = (0, $893a66ba513f4a66$exports.useBreakpoint)();
    let { direction: direction } = (0, $5XTxG$reactariai18n.useLocale)();
    let { matchedBreakpoints: matchedBreakpoints = (breakpointProvider === null || breakpointProvider === void 0 ? void 0 : breakpointProvider.matchedBreakpoints) || [
        'base'
    ] } = options;
    let styles = $d3b73be57066120b$export$f3c39bb9534218d0(props, handlers, direction, matchedBreakpoints);
    let style = {
        ...UNSAFE_style,
        ...styles
    };
    // @ts-ignore
    if (otherProps.className) console.warn("The className prop is unsafe and is unsupported in React Spectrum v3. Please use style props with Spectrum variables, or UNSAFE_className if you absolutely must do something custom. Note that this may break in future versions due to DOM structure changes.");
    // @ts-ignore
    if (otherProps.style) console.warn("The style prop is unsafe and is unsupported in React Spectrum v3. Please use style props with Spectrum variables, or UNSAFE_style if you absolutely must do something custom. Note that this may break in future versions due to DOM structure changes.");
    let styleProps = {
        style: style,
        className: UNSAFE_className
    };
    if ($d3b73be57066120b$export$52dbfdbe1b2c3541(props.isHidden, matchedBreakpoints)) styleProps.hidden = true;
    return {
        styleProps: styleProps
    };
}
function $d3b73be57066120b$export$46b6c81d11d2c30a(value) {
    return value;
}
function $d3b73be57066120b$export$52dbfdbe1b2c3541(prop, matchedBreakpoints) {
    if (prop && typeof prop === 'object' && !Array.isArray(prop)) {
        for(let i = 0; i < matchedBreakpoints.length; i++){
            let breakpoint = matchedBreakpoints[i];
            if (prop[breakpoint] != null) return prop[breakpoint];
        }
        return prop.base;
    }
    return prop;
}
